sdkproject/Assets/Mapbox/Unity/Map/QuadTreeTileProvider.cs

```diff
@@ -0,0 +1,193 @@
+﻿namespace Mapbox.Unity.Map
+{
+	using UnityEngine;
+	using Mapbox.Map;
+	using Mapbox.Unity.Utilities;
+	using Mapbox.Utils;
+	using System.Collections.Generic;
+	using System.Linq;
+	using System;
+
+	public class QuadTreeTileProvider : AbstractTileProvider
+	{
+		[SerializeField]
+		Camera _camera;
+
+
+		[SerializeField]
+		float _updateInterval;
+		Plane _groundPlane;
+		float _elapsedTime;
+		bool _shouldUpdate;
+
+		protected float _zoomRange;
+		public float ZoomRange
+		{
+			get
+			{
+				return _zoomRange;
+			}
+		}
+
+		public void SetZoomRange(float zoom)
+		{
+			_zoomRange = zoom;
+		}
+
+		protected Vector2d _panRange;
+		public Vector2d PanRange
+		{
+			get { return _panRange; }
+		}
+
+		public void SetPanRange(Vector2d pan, bool reset = false)
+		{
+			if (reset)
+			{
+				_mapPanned = false;
+				_panRange = Vector2d.zero;
+			}
+			else
+			{
+				_mapPanned = true;
+				_panRange = pan;
+			}
+		}
+
+		protected bool _mapPanned;
+		public bool MapPanned
+		{
+			get { return _mapPanned; }
+		}
+
+		public override void OnInitialized()
+		{
+			_groundPlane = new Plane(Vector3.up, 0);
+			_shouldUpdate = true;
+			_zoomRange = _map.Zoom;
+		}
+
+		public void UpdateMapProperties(Vector2d centerLatitudeLongitude, float zoom)
+		{
+			float differenceInZoom = 0.0f;
+			SetZoomRange(zoom);
+			if (Math.Abs(_map.Zoom - ZoomRange) > Constants.EpsilonFloatingPoint)
+			{
+				_map.SetZoom(zoom);
+				differenceInZoom = _map.Zoom - _map.InitialZoom;
+			}
+
+			//Update center latitude longitude
+			double xDelta = centerLatitudeLongitude.x;
+			double zDelta = centerLatitudeLongitude.y;
+
+			xDelta = xDelta > 0 ? Mathd.Min(xDelta, Mapbox.Utils.Constants.LatitudeMax) : Mathd.Max(xDelta, -Mapbox.Utils.Constants.LatitudeMax);
+			zDelta = zDelta > 0 ? Mathd.Min(zDelta, Mapbox.Utils.Constants.LongitudeMax) : Mathd.Max(zDelta, -Mapbox.Utils.Constants.LongitudeMax);
+
+			_map.SetCenterLatitudeLongitude(new Vector2d(xDelta, zDelta));
+			// Update the center based on current zoom level.
+			var referenceTileRect = Conversions.TileBounds(TileCover.CoordinateToTileId(_map.CenterLatitudeLongitude, _map.AbsoluteZoom));
+			_map.SetCenterMercator(referenceTileRect.Center);
```

Comment:
Why does the WebMerc center get set to the center of a tile?
Shouldn't this be:
```c#
_map.SetCenterMercator(Conversions.LatLonToMeters(centerLatitudeLongitude));
```